doc/devel/documenting_mpl.rst

```diff
@@ -7,12 +7,39 @@ Writing documentation
 Getting started
 ===============
 
+
+General file structure
+----------------------
+
+All documentation is built from the :file:`doc/` directory.  This directory
+contains both reStructuredText (ReST_; ``.rst``) files that contain pages in
+the documentation and configuration files for Sphinx_.
+
+The ``.rst`` files are kept in :file:`doc/users`,
+:file:`doc/devel`, :file:`doc/api` and :file:`doc/faq`. The main entry point is
+:file:`doc/index.rst`, which pulls in the :file:`index.rst` file for the users
+guide, developers guide, api reference, and FAQs. The documentation suite is
+built as a single document in order to make the most effective use of cross
+referencing.
+
+Sphinx_ also creates ``.rst`` files that are staged in :file:`doc/api` from
+the docstrings of the classes in the Matplotlib library.  Except for
+:file:`doc/api/api_changes/`, these ``.rst`` are created when the
+documentation is built.
+
+Similarly, the contents of :file:`docs/gallery` and :file:`docs/tutorials` are
+generated by the `Sphinx Gallery`_ from the sources in :file:`examples` and
+:file:`tutorials`.  These sources consist of python scripts that have ReST_
+documentation built into their comments.  Again, don't directly edit the
```

Comment:
This is the first time not editing ``.rst`` files is mentioned, maybe mention it earlier or get rid of "Again".